clients/src/test/java/org/apache/kafka/clients/consumer/internals/StreamsMembershipManagerTest.java

```diff
@@ -326,17 +340,233 @@ public void testReconcilingEmptyToMultipleActiveTaskOfDifferentSubtopologies() {
+
+        final Set<TopicPartition> expectedPartitionsToRevoke = Set.of(new TopicPartition(TOPIC_0, PARTITION_0));
+        verify(subscriptionState).markPendingRevocation(expectedPartitionsToRevoke);
+        onTasksRevokedCallbackExecuted.complete(null);
+        verify(subscriptionState).assignFromSubscribedAwaitingCallback(Collections.emptySet(), Collections.emptySet());
+        verifyInStateReconciling(membershipManager);
+        onTasksAssignedCallbackExecuted.complete(null);
+        verify(subscriptionState).enablePartitionsAwaitingCallback(Collections.emptySet());
+        verifyInStateAcknowledging(membershipManager);
+    }
+
+    @Test
+    public void testReconcilingStandbyTaskToActiveTask() {
+        setupStreamsAssignmentInterfaceWithOneSubtopologyOneSourceTopic(SUB_TOPOLOGY_ID_0, TOPIC_0);
+        final CompletableFuture<Void> onTasksAssignedCallbackExecutedSetup = new CompletableFuture<>();
+        final CompletableFuture<Void> onTasksAssignedCallbackExecuted = new CompletableFuture<>();
+        final Set<StreamsAssignmentInterface.TaskId> standbyTasksSetup = Set.of(
+            new StreamsAssignmentInterface.TaskId(SUB_TOPOLOGY_ID_0, PARTITION_0)
+        );
+        final Set<StreamsAssignmentInterface.TaskId> activeTasks = Set.of(
+            new StreamsAssignmentInterface.TaskId(SUB_TOPOLOGY_ID_0, PARTITION_0)
+        );
+        when(
+            streamsAssignmentInterface.requestOnTasksAssignedCallbackInvocation(
+                makeTaskAssignment(Collections.emptySet(), standbyTasksSetup)
+            )
+        ).thenReturn(onTasksAssignedCallbackExecutedSetup);
+        when(
+            streamsAssignmentInterface.requestOnTasksAssignedCallbackInvocation(
+                makeTaskAssignment(activeTasks, Collections.emptySet())
+            )
+        ).thenReturn(onTasksAssignedCallbackExecuted);
+        when(subscriptionState.assignedPartitions())
+            .thenReturn(Collections.emptySet())
+            .thenReturn(Collections.emptySet())
+            .thenReturn(Set.of(new TopicPartition(TOPIC_0, PARTITION_0)));
+        joining();
+        reconcile(makeHeartbeatResponseWithStandbyTasks(SUB_TOPOLOGY_ID_0, List.of(PARTITION_0)));
+        acknowledging(onTasksAssignedCallbackExecutedSetup);
+
+        reconcile(makeHeartbeatResponseWithActiveTasks(SUB_TOPOLOGY_ID_0, List.of(PARTITION_0)));
+
+        verify(subscriptionState).assignFromSubscribedAwaitingCallback(Collections.emptySet(), Collections.emptySet());
+        verifyInStateReconciling(membershipManager);
+        onTasksAssignedCallbackExecuted.complete(null);
+        verify(subscriptionState).enablePartitionsAwaitingCallback(Collections.emptySet());
+        verifyInStateAcknowledging(membershipManager);
+        verify(streamsAssignmentInterface, never()).requestOnTasksRevokedCallbackInvocation(any());
+        verify(subscriptionState, never()).markPendingRevocation(any());
+    }
+
```

Comment:
For all the new tests maybe consider one line comments for the major stages each test is running through, what the initial state is and the end state that is being validated.  One can get a sense from reading the code, but a quick comment would help with the understanding.